homeassistant/components/alert/__init__.py

```diff
@@ -103,23 +105,34 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
         title_template = cfg.get(CONF_TITLE)
         data = cfg.get(CONF_DATA)
 
-        entities.append(
-            Alert(
-                hass,
-                object_id,
-                name,
-                watched_entity_id,
-                alert_state,
-                repeat,
-                skip_first,
-                message_template,
-                done_message_template,
-                notifiers,
-                can_ack,
-                title_template,
-                data,
+        # Get entities based on the device_class
+        matching_entities = []
+        if device_class:
+            for entity_id in hass.states.async_entity_ids():
+                state = hass.states.get(entity_id)
+                if state is not None and state.attributes.get("device_class") == device_class:
+                    matching_entities.append(entity_id)
+        else:
+            matching_entities.append(watched_entity_id)
```

Comment:
This is not a feature we should add to integrations. Filtering capabilities like this, will quickly spiral out of control with dozens of variants.

Additionally, this has other flaw, as it relies on the set created at the point this integration and alert entity is initialized. As order of loading isn't guaranteed, it might miss the entities of an integration that is set up after this one.

That said, development on this integration is currently frozen and do not process changes for it, as the functionality of this integration can be achieved using an automation or automation blueprint. 

Because of the above reasons, I'll go ahead and close this PR. Nevertheless, thanks for being willing to contribute 👍 

../Frenck